include/alpaka/core/Common.hpp

```diff
@@ -80,13 +80,13 @@
 //! Macro defining the inline function attribute.
 //!
 //! The macro should stay on the left hand side of keywords, e.g. 'static', 'constexpr', 'explicit' or the return type.
-#if BOOST_LANG_CUDA || BOOST_LANG_HIP
+#if BOOST_LANG_CUDA
 #    define ALPAKA_FN_INLINE __forceinline__
 #elif BOOST_COMP_MSVC || defined(BOOST_COMP_MSVC_EMULATED)
 // TODO: With C++20 [[msvc::forceinline]] can be used.
 #    define ALPAKA_FN_INLINE __forceinline
 #else
-// For gcc, clang, and clang-based compilers like Intel icpx
+// For gcc, clang, and clang-based compilers like Intel icpx or hip-clang
 #    define ALPAKA_FN_INLINE [[gnu::always_inline]] inline
```

Comment:
Internally, HIP just transforms `__forceinline__` to `[[gnu::always_inline]]`, so we can just use this path directly.

Reply:
I am wondering about the new error in the CI:

```
/builds/hzdr/crp/alpaka/include/alpaka/meta/CudaVectorArrayWrapper.hpp:293:28: error: an attribute list cannot appear here
        ALPAKA_FN_HOST_ACC ALPAKA_FN_INLINE constexpr value_type& operator[]([[maybe_unused]] int const k) noexcept
                           ^~~~~~~~~~~~~~~~
/builds/hzdr/crp/alpaka/include/alpaka/core/Common.hpp:90:30: note: expanded from macro 'ALPAKA_FN_INLINE'
#    define ALPAKA_FN_INLINE [[gnu::always_inline]] inline
```

The attribute is the most left before are keys e.g. static, constexpr, ... I do not understand why this error is shown.

Reply:
In llvm I can only find for CUDA the definition of `__forceinline__`

```
#define __forceinline__ __device__ __inline__ __attribute__((always_inline))
```

For HIP this is not defined in llvm.
HIPRTC is in cmake defining
```
#define __forceinline__ inline __attribute__((always_inline))\n\
```

Reply:
No matter. I restored the old behaviour which *hopefully* makes this whole mess compile.